### ISSUE TITLE
add feature running

### ISSUE THREAD

**Issue by alistair-broomhead** [london-python-project-nights/romaine#5]

**Comment by geokala** [comment]
Take in feature object and retrieves located steps and runs each step of features.

**Comment by alistair-broomhead** [comment]
Should also warn and halt on unimplemented steps

**Comment by geokala** [comment]
Unimplemented steps presumably? Or do you mean that if a feature file is empty it should halt? (In that case I think a warning is the correct behaviour, but error+halt is... probably not good?).

**Comment by alistair-broomhead** [comment]
Yes steps sorry. I'm thinking this feature probably just provides a simple API for this stuff, and calling the API is left to the runner, making this a prerequisite for completing the runner feature. The other option is to make the runner feature a prerequisite for even starting this...

**Comment by geokala** [comment]
I thought this was the runner? Which issue are we talking about at the moment?

**Comment by alistair-broomhead** [comment]
Woops sorry, I was getting somewhat confused

**Comment by alistair-broomhead** [comment]
This will require #6

**Comment by alistair-broomhead** [comment]
So while working on this I've realised that our step-finding logic is not up to scratch, it doesn't allow for different step definitions for different step types, and it would perhaps be better to have it set up to be queried by step type
